File: Maxglass.Ecommerce/Maxglass.Ecommerce.DataTransfer/Produtos/Produtos/Requests/ProdutoListarRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Maxglass.Ecommerce.DataTransfer.Produtos.Produtos.Requests
{
    public class ProdutoListarRequest
    {
        public int? Tipo {get; set;}
        public string? Nome {get; set;}
        public int? Categoria {get; set;}
        public virtual string? Aro { get; set; }
        public virtual string? Borda { get; set; }
        public virtual string? Carcaca { get; set; }
        public virtual string? Lente { get; set; }
        public virtual int? Friso { get; set; }
        public virtual string? Posicao { get; set; }
        public string? Material { get; set; }
        public string? MaterialBorracha { get; set; }
        public int? AberturaFriso { get; set; }
        public string? Moldura { get; set; }
        public int? FuroEscapamento { get; set; }
        public int? AberturaSpoiler { get; set; }
        public string? Capa { get; set; }
        public int? PiscaAlerta { get; set; }
        public int? SensorPontoCego { get; set; }
        public string? Faixa { get; set; }
        public int? Marca { get; set; }
        
    }
}

File: Maxglass.Ecommerce.Web/src/app/produtos/models/produto-listar.request.ts
export class ProdutoListarRequest {
    tipo: number
    nome: string
    categoria: number
    aro: string
    borda: string
    carcaca: string
    lente: string
    friso: number
    posicao: string
    material: string
    materialBorracha: string
    aberturaFriso: number
    moldura: string
    furoEscapamento: number
    aberturaSpoiler: number
    capa: string
    piscaAlerta: number
    sensorPontoCego: number
    faixa: string
    marca: number

    constructor(params: Partial<ProdutoListarRequest>) {
        this.tipo = params.tipo || null;
        this.nome = params.nome || null;
        this.categoria = params.categoria || null;
        this.aro = params.aro || null;
        this.borda = params.borda || null;
        this.carcaca = params.carcaca || null;
        this.lente = params.lente || null;
        this.friso = params.friso || null;
        this.posicao = params.posicao || null;
        this.material = params.material || null;
        this.materialBorracha = params.materialBorracha || null;
        this.aberturaFriso = params.aberturaFriso || null;
        this.moldura = params.moldura || null;
        this.furoEscapamento = params.furoEscapamento || null;
        this.aberturaSpoiler = params.aberturaSpoiler || null;
        this.capa = params.capa || null;
        this.piscaAlerta = params.piscaAlerta || null;
        this.sensorPontoCego = params.sensorPontoCego || null;
        this.faixa = params.faixa || null;
        this.marca = params.marca || null;

    }
}
